Mathlib/Lean/Elab/Tactic/Basic.lean

```diff
@@ -18,4 +18,22 @@ Remark: note that `MVarId.getType'` uses `whnf` instead of `cleanupAnnotations`,
 def getMainTarget'' : TacticM Expr := do
   (← getMainGoal).getType''
 
+/--
+Like `focusAndDone` but takes a scope (e.g. tactic name) as an argument to
+produce more detailed error messages
+-/
+def focusAndDoneWithScope (scope : MessageData) (tactic : TacticM α) : TacticM α :=
+  focus do
+    -- TODO: Should add error message with the scope if the tactic (block) fails
+    let a ← tactic
```

Comment:
It would be an optional new feature to add the scope to the error message from the tactic block, but I don't know how to properly wrap it into a `try ... catch` block because of the `pure a` at the end.

Reply:
Here's a suggestion:
```lean
/--
Like `done` but takes a scope (e.g. a tactic name) as an argument
to produce more detailed error messages. -/
def doneWithScope (scope : MessageData) : TacticM Unit := do
  let gs ← getUnsolvedGoals
  unless gs.isEmpty do
    logError m!"{scope} failed to solve some goals.\n"
    Term.reportUnsolvedGoals gs
    throwAbortTactic

/--
Like `focusAndDone` but takes a scope (e.g. a tactic name) as an argument to
produce more detailed error messages.
-/
def focusAndDoneWithScope (scope : MessageData) (tactic : TacticM α) : TacticM α :=
  focus do
    let a ← try tactic catch e =>
      if isAbortTacticException e then throw e
      else throwError "{scope} failed.\n{← nestedExceptionToMessageData e}"
    doneWithScope scope
    pure a
```

Reply:
Thanks, lgtm!